src/modules/requirements/SpecialEventRandomRequirement.ts

```diff
@@ -0,0 +1,15 @@
+import SeededRand from '../utilities/SeededRand';
+import SpecialEventRequirement from './SpecialEventRequirement';
+
+export default class SpecialEventRandomRequirement extends SpecialEventRequirement {
+    private isAvailable = false;
+
+    constructor(specialEventName: string) {
+        super(specialEventName);
+        this.isAvailable = SeededRand.boolean();
+    }
+
+    public getProgress(): number {
+        return +(this.isAvailable && super.getProgress());
+    }
+}
```

Comment:
I think it would be worth adding special hint text for this as it's being added to every route with a requirement, maybe something along the lines of `The Pokémon seems to be avoiding the area this year` (just as an example).

Also curious as I haven't actually looked over the event calendar code, why does the previous code not work anymore?
As that only added it to routes based on the boolean rather than locking behind a req

Reply:
Something like that?

And yeah - event pokemon were moved to requirements to work with the event calendar. Halloween was the last hold out because of the unique distribution based on year